dask/bag/core.py

```diff
@@ -801,6 +801,10 @@ def to_dataframe(self, columns=None):
                             name, columns, divisions)
 
 
+normalize_token.register(Item, lambda a: a.key)
+normalize_token.register(Bag, lambda a: a.name)
```

Comment:
Using `base.tokenize` in imperative required that `normalize_token` knew how to handle all collections.